modules/widgets/package.json

```diff
@@ -30,7 +30,8 @@
     "prepublishOnly": "npm run build-bundle && npm run build-bundle -- --env=dev"
   },
   "dependencies": {
-    "@babel/runtime": "^7.0.0"
+    "@babel/runtime": "^7.0.0",
+    "preact": "^10.17.0"
```

Comment:
Just curious, I haven't really being paying attention. Are we using a react style library for our pure-js implementation? And then we will have react wrappers on top?

Reply:
I'm pretty certain we don't want the complexity or bundle size of react, but I like its api style.

My goal has been to pick something as close to the DOM as we can while also providing some convenience over `document.createElement`. `preact` seems to serve this niche well... It give us the ability to make basic components with JSX, and implements a thin virtual dom only adding 3kb to the bundle.

Re: react wrappers, I think of a widget as a deck component that happens to have UI. In a sense, how the UI is built within it is ideally isolated from other UI on the page so that deck can manage it. Let's say we used react to make UI widgets in `@deckgl/widgets`... I'd still ideally want them to [create their own react root](https://react.dev/learn/add-react-to-an-existing-project#using-react-for-a-part-of-your-existing-page) separate from the rest of application and other widgets. This way all integrations work through the same interface.

Reply:
Understood and I agree with the principle that widgets should be free to choose implementation as long as they are externally independent. That said, I would not be happy to see a pure-js deck.gl module have react as a dependency in its package.json. I would think twice before using that in a pure-js app.

For a community widget that could be a different story of course.

preact seems like a reasonable choice.